transfer/src/main/java/com/kin/ecosystem/transfer/AccountInfoService.java

```diff
@@ -0,0 +1,167 @@
+
+    @Override
+    protected void onHandleIntent(final @Nullable Intent intent) {
+        startForeground();
+
+        if (intent != null && intent.getAction().equals(ACTION_KIN_TRANSFER_INFO) && intent.hasExtra(EXTRA_ACTION_KEY)
+                && intent.hasExtra(EXTRA_PACKAGE_KEY) && intent.hasExtra(EXTRA_SERVICE_FULL_PATH_KEY)) {
+            getAccountInfo(new AccountInfoListener() {
+                @Override
+                public void onAccountInfoReady(String data) {
+                    if (createFile(data)) {
+                        startServiceKinAccountReceiver(intent.getStringExtra(EXTRA_ACTION_KEY),
+                                intent.getStringExtra(EXTRA_PACKAGE_KEY),
+                                intent.getStringExtra(EXTRA_SERVICE_FULL_PATH_KEY));
+                    }
+                }
+
+                @Override
+                public void onAccountInfoFailed(String reason) {
+                }
+            });
+
+        }
+        stopForeground();
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        fileDirPath = getFilesDir().getAbsolutePath() + File.separator + FILE_PROVIDER_DIR_NAME;
+        fileFullPath = fileDirPath + File.separator + FILE_NAME;
+    }
+
+    private void stopForeground() {
+        stopForeground(true);
+        stopSelf();
+    }
+
+    private void startForeground() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            Notification.Builder builder = new Notification.Builder(this, createNotificationChannel());
+            builder.setContentTitle(getString(R.string.app_name)).setContentText(NOTIFICATION_TRANSFER).setAutoCancel(true);
+            Notification notification = builder.build();
+            startForeground(1, notification);
+        } else {
+            NotificationCompat.Builder builder = new NotificationCompat.Builder(this);
+            builder.setContentTitle(getString(R.string.app_name)).setContentText(NOTIFICATION_TRANSFER).setPriority(NotificationCompat.PRIORITY_DEFAULT).setAutoCancel(true);
+            Notification notification = builder.build();
+            startForeground(1, notification);
+        }
+    }
+
+    private boolean createFile(String data) {
+        File dir = new File(fileDirPath);
+        if (!dir.exists()) {
+            if (!dir.mkdir()) {
+                return false;
+            }
+        }
+        try {
+            file = new File(fileFullPath);
+            if (!file.exists()) {
+                if (!file.createNewFile()) {
+                    return false;
+                }
+            }
+            FileWriter writer = new FileWriter(file);
+            writer.append(data);
+            writer.flush();
+            writer.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+            Log.e("AccountInfoService", "AccountInfoService cant create and write to file " + e.getMessage());
```

Comment:
we can remove `e.printStackTrace();`  and Log ?
The developer can recover from that ? get any notification ?